iotedgedev/deploymentmanifest.py

```diff
@@ -37,13 +37,37 @@ def add_module_template(self, module_name):
             "status": "running",
             "restartPolicy": "always",
             "settings": {
-              "image": \"{MODULES.""" + module_name + """.amd64}\",
+              "image": \"${MODULES.""" + module_name + """.amd64}\",
               "createOptions": ""
             }
         }"""
 
         self.json["moduleContent"]["$edgeAgent"]["properties.desired"]["modules"][module_name] = json.loads(new_module)
 
+        self.add_default_route(module_name)
+
+    def add_default_route(self, module_name):
+        """Add a default route to send messages to IoT Hub"""
+        new_route_name = "{0}ToIoTHub".format(module_name)
+        new_route = "FROM /messages/modules/{0}/outputs/* INTO $upstream".format(module_name)
+
+        self.json["moduleContent"]["$edgeHub"]["properties.desired"]["routes"][new_route_name] = new_route
```

Comment:
Seems like we should check if the values exist.  Thoughts on loaded the deployment file into memory object instead of json, and then serialize to json when done?  I think the interface to the deployment manifest would be simplified that way.

Reply:
@jongio I added a `nested_set` utility method to ensure that the keys exist when we set the values in a nested way.